### PR TITLE
Add missing Refund class

### PR DESCRIPTION
**Pull request by nwithan8**
Our C# client library is missing a `Refund` class. Let's fix that.

This PR includes:
- A new `Refund` class, with `Create()`, `Retrieve()` and `All()` functions
- Refund tests for the .NET/.NET Core (VCR) and .NET Framework (live) versions in the respective test suites.

This branch is based on the `tests` branch, so please merge the `tests` branch first to avoid merge conflicts.

### CODE REVIEW

**Comment by jchen293** [comment]
I think this PR should be reviewed after the Test suite is merged to the master branch, and then once you rebase from the master branch, the number of files changed will be much less.

**Comment by Justintime50** [comment]
Yeah this definitely needs a rebase. Drafting.

**Review by Justintime50 — APPROVED** [review]

File: EasyPost.Tests.Net/RefundTest.cs [hunk 1]
```diff
@@ -1,16 +1,93 @@
...
+        public void TestCreate()
+        {
+            VCR.Record("create");
+
+            List<Refund> refunds = CreateBasicRefund();
+
+            foreach (var item in refunds)
+            {
+                Assert.IsInstanceOfType(item, typeof(Refund));
+            }
```
Comment (jchen293) [inline EasyPost.Tests.Net/RefundTest.cs:44]:
```suggestion
            foreach (var refund in refunds)
            {
                Assert.IsInstanceOfType(refund, typeof(Refund));
            }
```

File: EasyPost.Tests.Net/RefundTest.cs [hunk 2]
```diff
@@ -1,16 +1,93 @@
...
+            });
+
+            List<Refund> refunds = refundCollection.refunds;
+
+            Assert.IsTrue(refunds.Count <= Fixture.PageSize);
+            Assert.IsNotNull(refundCollection.has_more);
+            foreach (var item in refunds)
+            {
+                Assert.IsInstanceOfType(item, typeof(Refund));
+            }
```
Comment (jchen293) [inline EasyPost.Tests.Net/RefundTest.cs:70]:
```suggestion
            foreach (var refund in refunds)
            {
                Assert.IsInstanceOfType(refund, typeof(Refund));
            }
```

File: EasyPost.Tests.NetFramework/RefundTest.cs [hunk 3]
```diff
@@ -1,16 +1,87 @@
...
+
+        [TestMethod]
+        public void TestCreate()
+        {
+            List<Refund> refunds = CreateBasicRefund();
+
+            foreach (var item in refunds)
+            {
+                Assert.IsInstanceOfType(item, typeof(Refund));
+            }
```
Comment (jchen293) [inline EasyPost.Tests.NetFramework/RefundTest.cs:42]:
```suggestion
            foreach (var refund in refunds)
            {
                Assert.IsInstanceOfType(refund, typeof(Refund));
            }
```

File: EasyPost.Tests.NetFramework/RefundTest.cs [hunk 4]
```diff
@@ -1,16 +1,87 @@
...
+            });
+
+            List<Refund> refunds = refundCollection.refunds;
+
+            Assert.IsTrue(refunds.Count <= Fixture.PageSize);
+            Assert.IsNotNull(refundCollection.has_more);
+            foreach (var item in refunds)
+            {
+                Assert.IsInstanceOfType(item, typeof(Refund));
+            }
```
Comment (jchen293) [inline EasyPost.Tests.NetFramework/RefundTest.cs:66]:
```suggestion
            foreach (var refund in refunds)
            {
                Assert.IsInstanceOfType(refund, typeof(Refund));
            }
```

**Review by jchen293 — APPROVED** [review]
Won't block it, just some readability suggestions.

**Pull request merged**